pallets/capacity/src/lib.rs

```diff
@@ -211,6 +226,36 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		// TODO: real weights
+		#[pallet::weight(T::WeightInfo::stake())]
+		/// removes all thawed UnlockChunks from caller's StakingAccount and unlocks the sum of the thawed values
+		/// in the caller's token account.
+		///
+		/// ### Errors
+		///   - Returns `Error::NotAStakingAccount` if no StakingAccountDetails are found for `origin`.
+		///
+		pub fn withdraw_unstaked(origin: OriginFor<T>) -> DispatchResult {
+			let staker = ensure_signed(origin)?;
+
+			let mut staking_account =
+				Self::get_staking_account_for(&staker).ok_or(Error::<T>::NotAStakingAccount)?;
+			let current_block = frame_system::Pallet::<T>::block_number();
+
+			let amount_withdrawn = staking_account.reap_thawed(current_block);
+			ensure!(!amount_withdrawn.is_zero(), Error::<T>::NoUnstakedTokensAvailable);
+
+			if staking_account.total.is_zero() {
```

Comment:
How do you feel about having a function `staking_account.can_delete_account` this would make this line read a little nicer.

Reply:
To be honest I dislike making single-line, single-use functions since it just adds another stack frame, and really, what are we getting out of it that a simple comment cannot straighten out, if needed?  If there is ever another place where we check if the staking account has a zero total balance, or if we add more checks before deleting the staking account, as well, then sure let's abstract it,